src/app/plugins/cat.plugin.ts

```diff
@@ -0,0 +1,143 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage } from '../../common/types';
+
+export class CatPlugin extends Plugin {
+  public name: string = 'Cat Plugin';
+  public description: string = 'Generates pictures of cats, via popular command.';
+  public usage: string = 'cat breed or pic bomb(optional) <breed (optional)>';
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://api.thecatapi.com/v1/';
+  private _breeds: string[] = [];
+  private _breedsID: string[] = [];
+
+  constructor(public container: IContainer) {
+    super();
+    this.container.httpService
+      .get(`${this._API_URL}breeds`)
+      .then((response: IHttpResponse) => {
+        const breeds = response.data;
+
+        for (let index = 0; index < breeds.length; index++) {
+            const breed = breeds[index].name;
+            const breedID = breeds[index].id;
+            this._breeds.push(breed.toLowerCase())
+            this._breedsID.push(breedID.toLowerCase())
```

Comment:
Instead, keep each breed as a pair of a name and an ID to strongly associate each name with its ID.